privacyidea/lib/security/default.py

```diff
@@ -113,30 +113,111 @@ def decrypt(self, value, iv=None):
                   "the method : %s " % (fname,))
         raise NotImplementedError("Should have been implemented {0!s}".format(fname))
 
-    ''' higer level methods '''
-    def encrypt_password(self, clear_pass):
-        fname = 'encrypt_password'
-        log.error("This is the base class. You should implement "
-                  "the method : %s " % (fname,))
-        raise NotImplementedError("Should have been implemented {0!s}".format(fname))
+    def decrypt_password(self, crypt_pass):
+        """
+        Decrypt the given password. The CONFIG_KEY is used to decrypt it.
 
-    def encrypt_pin(self, clear_pin):
-        fname = 'encrypt_pin'
-        log.error("This is the base class. You should implement "
-                  "the method : %s " % (fname,))
-        raise NotImplementedError("Should have been implemented {0!s}".format(fname))
+        :param crypt_pass: the encrypted password with the leading iv,
+            separated by the ':', hexlified
+        :param crypt_pass: str
 
-    def decrypt_password(self, crypt_pass):
-        fname = 'decrypt_password'
-        log.error("This is the base class. You should implement "
-                  "the method : %s " % (fname,))
-        raise NotImplementedError("Should have been implemented {0!s}".format(fname))
+        :return: decrypted data
+        :rtype: str
+        """
+        return self._decrypt_value(crypt_pass, self.CONFIG_KEY)
 
     def decrypt_pin(self, crypt_pin):
-        fname = 'decrypt_pin'
-        log.error("This is the base class. You should implement "
-                  "the method : %s " % (fname,))
-        raise NotImplementedError("Should have been implemented {0!s}".format(fname))
+        """
+        Decrypt the given encrypted PIN with the TOKEN_KEY
+
+        :param crypt_pin: the encrypted pin with the leading iv,
+            separated by the ':'
+        :param crypt_pin: byte string
```

Comment:
Do You mean `:type crypt_pin: str`

Reply:
Ah look there - copy and paste.
Will fix this.

Reply:
done in 9baf77b7